types/es-abstract/helpers/regexTester.d.ts

```diff
@@ -0,0 +1,2 @@
+declare function regexTester(regex: RegExp): typeof RegExp.prototype.test;
```

Comment:
this type isn't accurate; it returns a call-bound `RegExp.prototype.test`

Reply:
That’s not really describable using **TypeScript**’s type system.

Reply:
ok, but that doesn't make `typeof RegExp.prototype.test` correct :-) what about:
```ts
(string) => ReturnType<RegExp.prototype.test>
```
?

Reply:
I inlined the return type of `callBind<typeof RegExp.prototype.test>`.

Reply:
you might be able to use a `this: undefined` parameter to represent a call-bound method (or some other type if I misunderstood what call-bound means).

Reply:
@sandersn call-bound means that the first argument becomes the `this` of the bound method, ie, `callBoundFoo(a, b, c)` is like `foo.call(a, b, c)` but without relying on `Function.prototype.call` existing.

Reply:
Call‑bound in this case means:
```ts
regexTester(regex: RegExp) {
	return Function.prototype.call.bind(RegExp.prototype.test, regex);
}
```